### PR TITLE
US #4302: Improve tagging system. Use generic colors for porject tags

### PR DESCRIPTION
**Pull request by bameda**

### CODE REVIEW

**Comment by Xaviju** [comment]
There are some mistakes according to UX in this PR:

- [ ] A user cannot change the tag color in the tag list without clicking in the edition mode.
- [ ] Missing transparent color option
- [ ] If the tag search filter shows empty results, display message
- [ ] Add new tag appears at the bottom of the list instead of the top.
- [ ] A user cannot drag and drop tags.
- [ ] Merge button should only appear if there are tags selected.
- [ ] When pressing on merge it just becomes merged but the UX shows some transition. We should decide how that transition is made but currently is hard to follow what happens.

**Comment by jespino** [comment]
The user can't reorder the tags currently, so drag and drop doesn't must work

**Comment by jespino** [comment]
The new tags must appear in the correct position based in the alphabetical order, because the order isn't custom.

**Comment by jespino** [comment]
The add form of new tags appear at the end of the tags because this is the behavior in the rest of the admin panel.

**Comment by Xaviju** [comment]
@jespino If I click on _merge tags_ the empty color gets lost and become gray.

**Comment by Xaviju** [comment]
There is only one test left to PR this. assigned to @juanfran

**Comment by Xaviju** [comment]
@juanfran p[lease check this (ask me if you need to) it would be nice to merge this since I need a component in this PR

**Comment by superalex** [comment]
I've been testing the admin area and I've detected two errors:
[] when creating a new tag if it exists the API returns error but the front doesn't display anything
[] if the project doesn't have any tag and I click on "create tag" no form is shown

**Comment by bameda** [comment]
I fix this two a some more minor errors in master branch
